deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/security/UserDataHelper.java

```diff
@@ -82,8 +84,15 @@ public String getUsername(String token) throws Exception {
         Map<String, String> params = ImmutableMap.of("access_token", token);
         String jsonPayload = httpClient.get(userDataUrl, params, null, 3);
         JsonObject jsonObject = new JsonParser().parse(jsonPayload).getAsJsonObject();
-        jsonObject = jsonObject.getAsJsonObject("user");
-        String user = jsonObject.get("username").getAsString();
+        final String user;
+        if (oauthProvider.equals("googlev2")){
```

Comment:
I'm thinking this should be the default. No need to check if "googlev2" or any other string at this time. Google follows the OpenID Connect format for UserInfo responses so this code would apply to other sources of providers.

Im summary:
if pinadmin: get username
else: get by email